core/internal/settings/settings.go

```diff
@@ -235,6 +240,61 @@ func (s *Settings) GetProgram() string {
 	return s.Proto.Program.GetValue()
 }
 
+// The relative path to the script that created the run, if available.
+func (s *Settings) GetProgramRelpath() string {
+	return s.Proto.ProgramRelpath.GetValue()
+}
+
+// The absolute path to the script that created the run, if available.
+func (s *Settings) GetProgramAbspath() string {
+	return s.Proto.ProgramAbspath.GetValue()
+}
+
+// The arguments passed to the script that created the run, if available.
+func (s *Settings) GetArgs() []string {
+	return s.Proto.XArgs.GetValue()
+}
+
+// The local path to the code directory, if available.
+func (s *Settings) GetCodePathLocal() string {
+	return s.Proto.XCodePathLocal.GetValue()
+}
+
+// The operating system of the machine running the run.
+func (s *Settings) GetOs() string {
```

Comment:
should use Go capitalization conventions:

```suggestion
func (s *Settings) GetOS() string {
```

Reply:
oh i think i changed it at some point lol, will fix it again